tvtk/special_gen.py

```diff
@@ -319,8 +319,11 @@ def __iter__(self):
                 yield obj.GetPoint(i)
 
         def _check_key(self, key, n):
-            if type(key) != type(1):
-                raise TypeError, "Only integers are valid keys."
+            ##############################################
+            # Allow int and long keys. Fixes GH Issue 173.
+            ##############################################
+            if type(key) not in [type(1), type(1L)]:
```

Comment:
If key is not a numeric value it will fail anyway in the comparisons that follow.